src/content/en/fundamentals/engage-and-retain/push-notifications/sending-messages.markdown

```diff
@@ -3,93 +3,343 @@ layout: shared/narrow
+samples frequently and ideally should be. Given the complexity of this subject, 
+we felt it was better to describe it at a conceptual level and restrict sample 
+codes to snipits that illustrate those concepts.
+
+### The product
+
+Let's look at what we're going to end up with, then we'll talk about how to 
+build it. 
+
+<pre>POST /push-service/send/dbDqU8xX10w:APA91b... HTTP/1.1  
+Host: push.example.net  
+Push-Receipt: https://push.example.net/r/3ZtI4YVNBnUUZhuoChl6omU  
+TTL: 43200  
+Content-Type: text/plain;charset=utf8  
+Content-Length: 36  
+Authorization: Bearer 
+eyJ0eXAiOiJKV1QiLCJhbGciOiJFUzI1NiJ9.eyJhdWQiOiJodHRwczovL3B1c2guZXhhbXBsZS5uZXQiLCJleHAiOjE0NTM1MjM3NjgsInN1YiI6Im1haWx0bzpwdXNoQGV4YW1wbGUuY29tIn0.i3CYb7t4xfxCDquptFOepC9GAu\_HLGkMlMuCGSK2rpiUfnK9ojFwDXb1JrErtmysazNjjvW2L9OkSSHzvoD1oA  
+Crypto-Key: 
+p256ecdsa=BA1Hxzyi1RUM1b5wjxsn7nGxAszw2u61m164i3MrAIxHF6YK5h4SDYic-dRuU\_RCPCfA5aq9ojSwk5Y2EmClBPsiChYuI3jMzt3ir20P8r\_jgRR-dSuN182x7iB</pre>
+
+It should be fairly obvious how the endpoint of the subscription object is used. 
+The Authorization, Crypto-Key , and TTL headers take a little explanation. Let's 
+start with the simplest.
+
+## HTTP headers
 
-_endpoint_—Contains two parts: the URL of the messaging service you're using
-followed by unique identifier for the user. This is called a subscription ID or
-a registration ID. This tells your server how to identify you the messaging
-server.
+### TTL
 
-_keys_—Encryption keys used for encrypting data passed to the service worker
-messages.
+It may be some time before a message server can deliver a message sent by your 
+app server. Message services aren't obligated to hold a message forever. 
+Frankly, to be timely, an app server should never send a message that can live 
+forever. That's why it's required to include a header called TTL, literally 
+'time to live'.
 
-## Encrypting a message {#encrypting}
+The TTL header is a a value in seconds given as a suggestion to the message 
+server for how long the message should be retained. If it chooses, the message 
+server may retain the message for less time than is given in the TTL header. If 
+it does, it must return that time in a TTL header in a response to the message 
+request. 
 
-You must encrypt all messages before sending them. Encryption is enough of a
-specialty, even within software development, that we don't recommend writing
-your own encryption system. Fortunately, there are [a range of push
-libraries](https://github.com/web-push-libs) including our own [Push Encryption
-Library](https://github.com/GoogleChrome/web-push-encryption).
+You can almost think of this interaction as a conversation. 
 
-## Two ways to send a push message {#ways-to-send}
+App server: "Here's a message can you deliver it within 43200 seconds?"
 
-When implementing server code there are two methods of sending messages to
-consider.
+Message server: "I really can't, but I can do it for you in 21600."
```

Comment:
Slight reword:

"The TTL header is a a value in seconds given as a suggestion to the message
server for how long the server should keep the message and attempt to deliver it. If it chooses, the message server may shorten the time it is willing to keep the message. If
it does, it must return that shortened time in a TTL header in a response to the message
request. If the TTL has a value of 0, the message server should deliver it
immediately if the user agent is available. If the user agent isn't available
the message expires immediately and is never delivered."